index/internal/collections/collections.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2025 EngFlow Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collections
+
+import (
+	"slices"
+)
+
+// Map applies the provided transformation function `fn` to each element of the input slice `ts`
+// and returns a new slice of the resulting values.
+//
+// Example:
```

Comment:
You might be interested to know that examples in Go can be [automatically run by the testing framework](https://pkg.go.dev/testing#hdr-Examples). They're specially rendered by the documentation viewer (pkg.go.dev) and can be exported into the playground.

No real need to do that for an internal library like this, but it's nice for anything externally facing.